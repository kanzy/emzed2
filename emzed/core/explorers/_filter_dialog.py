# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'filter_dialog.ui'
#
# Created by: PyQt4 UI code generator 4.11.4
#
# WARNING! All changes made in this file will be lost!

from PyQt4 import QtCore, QtGui

try:
    _fromUtf8 = QtCore.QString.fromUtf8
except AttributeError:
    def _fromUtf8(s):
        return s

try:
    _encoding = QtGui.QApplication.UnicodeUTF8
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig, _encoding)
except AttributeError:
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig)

class Ui_FilterDialog(object):
    def setupUi(self, FilterDialog):
        FilterDialog.setObjectName(_fromUtf8("FilterDialog"))
        FilterDialog.resize(480, 640)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Maximum, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(1)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(FilterDialog.sizePolicy().hasHeightForWidth())
        FilterDialog.setSizePolicy(sizePolicy)
        FilterDialog.setMinimumSize(QtCore.QSize(0, 0))
        self.verticalLayout = QtGui.QVBoxLayout(FilterDialog)
        self.verticalLayout.setMargin(5)
        self.verticalLayout.setObjectName(_fromUtf8("verticalLayout"))
        self.frame = QtGui.QFrame(FilterDialog)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Maximum, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.frame.sizePolicy().hasHeightForWidth())
        self.frame.setSizePolicy(sizePolicy)
        self.frame.setMinimumSize(QtCore.QSize(400, 0))
        self.frame.setObjectName(_fromUtf8("frame"))
        self.widgets = QtGui.QGridLayout(self.frame)
        self.widgets.setContentsMargins(0, 5, 3, 3)
        self.widgets.setSpacing(0)
        self.widgets.setObjectName(_fromUtf8("widgets"))
        self.groupBox = QtGui.QGroupBox(self.frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Maximum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.groupBox.sizePolicy().hasHeightForWidth())
        self.groupBox.setSizePolicy(sizePolicy)
        self.groupBox.setTitle(_fromUtf8(""))
        self.groupBox.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.groupBox.setFlat(True)
        self.groupBox.setObjectName(_fromUtf8("groupBox"))
        self.horizontalLayout = QtGui.QHBoxLayout(self.groupBox)
        self.horizontalLayout.setMargin(0)
        self.horizontalLayout.setObjectName(_fromUtf8("horizontalLayout"))
        spacerItem = QtGui.QSpacerItem(40, 20, QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        self.horizontalLayout.addItem(spacerItem)
        self.cancel_button = QtGui.QPushButton(self.groupBox)
        self.cancel_button.setFocusPolicy(QtCore.Qt.NoFocus)
        self.cancel_button.setObjectName(_fromUtf8("cancel_button"))
        self.horizontalLayout.addWidget(self.cancel_button)
        self.submit_button = QtGui.QPushButton(self.groupBox)
        self.submit_button.setFocusPolicy(QtCore.Qt.NoFocus)
        self.submit_button.setObjectName(_fromUtf8("submit_button"))
        self.horizontalLayout.addWidget(self.submit_button)
        self.widgets.addWidget(self.groupBox, 4, 0, 1, 1)
        self.expert_mode_box = QtGui.QGroupBox(self.frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.MinimumExpanding, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.expert_mode_box.sizePolicy().hasHeightForWidth())
        self.expert_mode_box.setSizePolicy(sizePolicy)
        self.expert_mode_box.setFlat(True)
        self.expert_mode_box.setCheckable(True)
        self.expert_mode_box.setChecked(False)
        self.expert_mode_box.setObjectName(_fromUtf8("expert_mode_box"))
        self.verticalLayout_2 = QtGui.QVBoxLayout(self.expert_mode_box)
        self.verticalLayout_2.setMargin(12)
        self.verticalLayout_2.setSpacing(1)
        self.verticalLayout_2.setObjectName(_fromUtf8("verticalLayout_2"))
        self.filter_expression = QtGui.QPlainTextEdit(self.expert_mode_box)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.filter_expression.sizePolicy().hasHeightForWidth())
        self.filter_expression.setSizePolicy(sizePolicy)
        self.filter_expression.setMinimumSize(QtCore.QSize(0, 120))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("Courier"))
        self.filter_expression.setFont(font)
        self.filter_expression.setFrameShape(QtGui.QFrame.NoFrame)
        self.filter_expression.setFrameShadow(QtGui.QFrame.Plain)
        self.filter_expression.setLineWidth(1)
        self.filter_expression.setObjectName(_fromUtf8("filter_expression"))
        self.verticalLayout_2.addWidget(self.filter_expression)
        self.widgets.addWidget(self.expert_mode_box, 1, 0, 1, 1)
        self.scrollArea = QtGui.QScrollArea(self.frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(5)
        sizePolicy.setHeightForWidth(self.scrollArea.sizePolicy().hasHeightForWidth())
        self.scrollArea.setSizePolicy(sizePolicy)
        self.scrollArea.setMinimumSize(QtCore.QSize(0, 0))
        self.scrollArea.setFrameShape(QtGui.QFrame.NoFrame)
        self.scrollArea.setFrameShadow(QtGui.QFrame.Plain)
        self.scrollArea.setHorizontalScrollBarPolicy(QtCore.Qt.ScrollBarAlwaysOff)
        self.scrollArea.setWidgetResizable(False)
        self.scrollArea.setAlignment(QtCore.Qt.AlignHCenter|QtCore.Qt.AlignTop)
        self.scrollArea.setObjectName(_fromUtf8("scrollArea"))
        self.scrollAreaWidgetContents = QtGui.QWidget()
        self.scrollAreaWidgetContents.setGeometry(QtCore.QRect(0, 0, 397, 423))
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(5)
        sizePolicy.setHeightForWidth(self.scrollAreaWidgetContents.sizePolicy().hasHeightForWidth())
        self.scrollAreaWidgetContents.setSizePolicy(sizePolicy)
        self.scrollAreaWidgetContents.setObjectName(_fromUtf8("scrollAreaWidgetContents"))
        self.filters_layout = QtGui.QGridLayout(self.scrollAreaWidgetContents)
        self.filters_layout.setContentsMargins(10, 15, 10, 20)
        self.filters_layout.setHorizontalSpacing(5)
        self.filters_layout.setVerticalSpacing(7)
        self.filters_layout.setObjectName(_fromUtf8("filters_layout"))
        self.scrollArea.setWidget(self.scrollAreaWidgetContents)
        self.widgets.addWidget(self.scrollArea, 0, 0, 1, 1)
        self.verticalLayout.addWidget(self.frame)

        self.retranslateUi(FilterDialog)
        QtCore.QObject.connect(self.expert_mode_box, QtCore.SIGNAL(_fromUtf8("toggled(bool)")), self.scrollArea.setDisabled)
        QtCore.QMetaObject.connectSlotsByName(FilterDialog)

    def retranslateUi(self, FilterDialog):
        FilterDialog.setWindowTitle(_translate("FilterDialog", "Dialog", None))
        self.cancel_button.setText(_translate("FilterDialog", "Cancel", None))
        self.submit_button.setText(_translate("FilterDialog", "Submit", None))
        self.expert_mode_box.setTitle(_translate("FilterDialog", "expert mode", None))


class FilterDialog(QtGui.QDialog, Ui_FilterDialog):
    def __init__(self, parent=None, f=QtCore.Qt.WindowFlags()):
        QtGui.QDialog.__init__(self, parent, f)

        self.setupUi(self)


if __name__ == "__main__":
    import sys
    app = QtGui.QApplication(sys.argv)
    FilterDialog = QtGui.QDialog()
    ui = Ui_FilterDialog()
    ui.setupUi(FilterDialog)
    FilterDialog.show()
    sys.exit(app.exec_())

