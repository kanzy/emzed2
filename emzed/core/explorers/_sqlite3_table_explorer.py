# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'sqlite3_table_explorer.ui'
#
# Created by: PyQt4 UI code generator 4.11.4
#
# WARNING! All changes made in this file will be lost!

from PyQt4 import QtCore, QtGui

try:
    _fromUtf8 = QtCore.QString.fromUtf8
except AttributeError:
    def _fromUtf8(s):
        return s

try:
    _encoding = QtGui.QApplication.UnicodeUTF8
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig, _encoding)
except AttributeError:
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig)

class Ui__Sqlite3TableExplorer(object):
    def setupUi(self, _Sqlite3TableExplorer):
        _Sqlite3TableExplorer.setObjectName(_fromUtf8("_Sqlite3TableExplorer"))
        _Sqlite3TableExplorer.resize(885, 609)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(_Sqlite3TableExplorer.sizePolicy().hasHeightForWidth())
        _Sqlite3TableExplorer.setSizePolicy(sizePolicy)
        _Sqlite3TableExplorer.setSizeGripEnabled(True)
        self.verticalLayout = QtGui.QVBoxLayout(_Sqlite3TableExplorer)
        self.verticalLayout.setContentsMargins(12, 5, 12, 12)
        self.verticalLayout.setObjectName(_fromUtf8("verticalLayout"))
        self.splitter = QtGui.QSplitter(_Sqlite3TableExplorer)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.splitter.sizePolicy().hasHeightForWidth())
        self.splitter.setSizePolicy(sizePolicy)
        self.splitter.setOrientation(QtCore.Qt.Vertical)
        self.splitter.setOpaqueResize(True)
        self.splitter.setObjectName(_fromUtf8("splitter"))
        self.plot_frame = QtGui.QFrame(self.splitter)
        self.plot_frame.setEnabled(True)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.plot_frame.sizePolicy().hasHeightForWidth())
        self.plot_frame.setSizePolicy(sizePolicy)
        self.plot_frame.setMinimumSize(QtCore.QSize(0, 250))
        self.plot_frame.setMaximumSize(QtCore.QSize(16777215, 16777215))
        self.plot_frame.setFrameShape(QtGui.QFrame.NoFrame)
        self.plot_frame.setFrameShadow(QtGui.QFrame.Raised)
        self.plot_frame.setObjectName(_fromUtf8("plot_frame"))
        self.horizontalLayout = QtGui.QHBoxLayout(self.plot_frame)
        self.horizontalLayout.setObjectName(_fromUtf8("horizontalLayout"))
        self.splitter_2 = QtGui.QSplitter(self.plot_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(1)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.splitter_2.sizePolicy().hasHeightForWidth())
        self.splitter_2.setSizePolicy(sizePolicy)
        self.splitter_2.setOrientation(QtCore.Qt.Horizontal)
        self.splitter_2.setObjectName(_fromUtf8("splitter_2"))
        self.eic_plotter = EicPlottingWidget(self.splitter_2)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(22)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.eic_plotter.sizePolicy().hasHeightForWidth())
        self.eic_plotter.setSizePolicy(sizePolicy)
        self.eic_plotter.setMinimumSize(QtCore.QSize(100, 100))
        self.eic_plotter.setObjectName(_fromUtf8("eic_plotter"))
        self.frame = QtGui.QFrame(self.splitter_2)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(1)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.frame.sizePolicy().hasHeightForWidth())
        self.frame.setSizePolicy(sizePolicy)
        self.frame.setMinimumSize(QtCore.QSize(0, 150))
        self.frame.setFrameShape(QtGui.QFrame.NoFrame)
        self.frame.setFrameShadow(QtGui.QFrame.Plain)
        self.frame.setObjectName(_fromUtf8("frame"))
        self.verticalLayout_3 = QtGui.QVBoxLayout(self.frame)
        self.verticalLayout_3.setSizeConstraint(QtGui.QLayout.SetMinimumSize)
        self.verticalLayout_3.setObjectName(_fromUtf8("verticalLayout_3"))
        self.integration_widget = IntegrationWidget(self.frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.integration_widget.sizePolicy().hasHeightForWidth())
        self.integration_widget.setSizePolicy(sizePolicy)
        self.integration_widget.setMinimumSize(QtCore.QSize(0, 0))
        self.integration_widget.setObjectName(_fromUtf8("integration_widget"))
        self.verticalLayout_3.addWidget(self.integration_widget)
        self.choose_spectra_widget = ChooseSpectraWidget(self.frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.choose_spectra_widget.sizePolicy().hasHeightForWidth())
        self.choose_spectra_widget.setSizePolicy(sizePolicy)
        self.choose_spectra_widget.setMinimumSize(QtCore.QSize(0, 0))
        self.choose_spectra_widget.setBaseSize(QtCore.QSize(0, 100))
        self.choose_spectra_widget.setObjectName(_fromUtf8("choose_spectra_widget"))
        self.verticalLayout_3.addWidget(self.choose_spectra_widget)
        spacerItem = QtGui.QSpacerItem(20, 0, QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Expanding)
        self.verticalLayout_3.addItem(spacerItem)
        self.mz_plotter = MzPlottingWidget(self.splitter_2)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(22)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.mz_plotter.sizePolicy().hasHeightForWidth())
        self.mz_plotter.setSizePolicy(sizePolicy)
        self.mz_plotter.setMinimumSize(QtCore.QSize(100, 100))
        self.mz_plotter.setObjectName(_fromUtf8("mz_plotter"))
        self.horizontalLayout.addWidget(self.splitter_2)
        self.table_view = QtGui.QTableView(self.splitter)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.MinimumExpanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(5)
        sizePolicy.setHeightForWidth(self.table_view.sizePolicy().hasHeightForWidth())
        self.table_view.setSizePolicy(sizePolicy)
        self.table_view.setMinimumSize(QtCore.QSize(0, 0))
        self.table_view.setLocale(QtCore.QLocale(QtCore.QLocale.English, QtCore.QLocale.UnitedStates))
        self.table_view.setFrameShape(QtGui.QFrame.NoFrame)
        self.table_view.setFrameShadow(QtGui.QFrame.Plain)
        self.table_view.setLineWidth(0)
        self.table_view.setAutoScroll(True)
        self.table_view.setDragEnabled(True)
        self.table_view.setAlternatingRowColors(True)
        self.table_view.setSelectionBehavior(QtGui.QAbstractItemView.SelectRows)
        self.table_view.setVerticalScrollMode(QtGui.QAbstractItemView.ScrollPerItem)
        self.table_view.setHorizontalScrollMode(QtGui.QAbstractItemView.ScrollPerPixel)
        self.table_view.setGridStyle(QtCore.Qt.NoPen)
        self.table_view.setSortingEnabled(False)
        self.table_view.setCornerButtonEnabled(True)
        self.table_view.setObjectName(_fromUtf8("table_view"))
        self.table_view.horizontalHeader().setCascadingSectionResizes(False)
        self.table_view.horizontalHeader().setMinimumSectionSize(100)
        self.table_view.horizontalHeader().setStretchLastSection(True)
        self.buttons_frame = QtGui.QFrame(self.splitter)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.buttons_frame.sizePolicy().hasHeightForWidth())
        self.buttons_frame.setSizePolicy(sizePolicy)
        self.buttons_frame.setMinimumSize(QtCore.QSize(0, 50))
        self.buttons_frame.setFrameShape(QtGui.QFrame.NoFrame)
        self.buttons_frame.setFrameShadow(QtGui.QFrame.Plain)
        self.buttons_frame.setObjectName(_fromUtf8("buttons_frame"))
        self.gridLayout = QtGui.QGridLayout(self.buttons_frame)
        self.gridLayout.setMargin(0)
        self.gridLayout.setObjectName(_fromUtf8("gridLayout"))
        self.second_sort_order = QtGui.QComboBox(self.buttons_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.second_sort_order.sizePolicy().hasHeightForWidth())
        self.second_sort_order.setSizePolicy(sizePolicy)
        self.second_sort_order.setObjectName(_fromUtf8("second_sort_order"))
        self.second_sort_order.addItem(_fromUtf8(""))
        self.second_sort_order.addItem(_fromUtf8(""))
        self.gridLayout.addWidget(self.second_sort_order, 0, 10, 1, 1)
        self.first_sort_order = QtGui.QComboBox(self.buttons_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.first_sort_order.sizePolicy().hasHeightForWidth())
        self.first_sort_order.setSizePolicy(sizePolicy)
        self.first_sort_order.setObjectName(_fromUtf8("first_sort_order"))
        self.first_sort_order.addItem(_fromUtf8(""))
        self.first_sort_order.addItem(_fromUtf8(""))
        self.gridLayout.addWidget(self.first_sort_order, 0, 7, 1, 1)
        spacerItem1 = QtGui.QSpacerItem(20, 20, QtGui.QSizePolicy.Fixed, QtGui.QSizePolicy.Minimum)
        self.gridLayout.addItem(spacerItem1, 0, 3, 1, 1)
        self.reset_filter_button = QtGui.QPushButton(self.buttons_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.reset_filter_button.sizePolicy().hasHeightForWidth())
        self.reset_filter_button.setSizePolicy(sizePolicy)
        self.reset_filter_button.setAutoDefault(False)
        self.reset_filter_button.setObjectName(_fromUtf8("reset_filter_button"))
        self.gridLayout.addWidget(self.reset_filter_button, 0, 2, 1, 1)
        self.second_sort_field = QtGui.QComboBox(self.buttons_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.MinimumExpanding, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.second_sort_field.sizePolicy().hasHeightForWidth())
        self.second_sort_field.setSizePolicy(sizePolicy)
        self.second_sort_field.setObjectName(_fromUtf8("second_sort_field"))
        self.gridLayout.addWidget(self.second_sort_field, 0, 9, 1, 1)
        self.label = QtGui.QLabel(self.buttons_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.label.sizePolicy().hasHeightForWidth())
        self.label.setSizePolicy(sizePolicy)
        self.label.setObjectName(_fromUtf8("label"))
        self.gridLayout.addWidget(self.label, 0, 5, 1, 1)
        self.first_sort_field = QtGui.QComboBox(self.buttons_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.MinimumExpanding, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.first_sort_field.sizePolicy().hasHeightForWidth())
        self.first_sort_field.setSizePolicy(sizePolicy)
        self.first_sort_field.setFrame(False)
        self.first_sort_field.setObjectName(_fromUtf8("first_sort_field"))
        self.gridLayout.addWidget(self.first_sort_field, 0, 6, 1, 1)
        self.visible_columns_button = QtGui.QPushButton(self.buttons_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.visible_columns_button.sizePolicy().hasHeightForWidth())
        self.visible_columns_button.setSizePolicy(sizePolicy)
        self.visible_columns_button.setFocusPolicy(QtCore.Qt.TabFocus)
        self.visible_columns_button.setAutoDefault(False)
        self.visible_columns_button.setDefault(False)
        self.visible_columns_button.setFlat(False)
        self.visible_columns_button.setObjectName(_fromUtf8("visible_columns_button"))
        self.gridLayout.addWidget(self.visible_columns_button, 0, 0, 1, 1)
        self.filter_button = QtGui.QPushButton(self.buttons_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.filter_button.sizePolicy().hasHeightForWidth())
        self.filter_button.setSizePolicy(sizePolicy)
        self.filter_button.setAutoDefault(False)
        self.filter_button.setDefault(False)
        self.filter_button.setObjectName(_fromUtf8("filter_button"))
        self.gridLayout.addWidget(self.filter_button, 0, 1, 1, 1)
        self.label_2 = QtGui.QLabel(self.buttons_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.label_2.sizePolicy().hasHeightForWidth())
        self.label_2.setSizePolicy(sizePolicy)
        self.label_2.setObjectName(_fromUtf8("label_2"))
        self.gridLayout.addWidget(self.label_2, 0, 8, 1, 1)
        self.current_filter = QtGui.QTextEdit(self.buttons_frame)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(1)
        sizePolicy.setHeightForWidth(self.current_filter.sizePolicy().hasHeightForWidth())
        self.current_filter.setSizePolicy(sizePolicy)
        self.current_filter.setMinimumSize(QtCore.QSize(0, 30))
        self.current_filter.setMaximumSize(QtCore.QSize(16777215, 30))
        font = QtGui.QFont()
        font.setFamily(_fromUtf8("Courier New"))
        self.current_filter.setFont(font)
        self.current_filter.setFrameShadow(QtGui.QFrame.Plain)
        self.current_filter.setVerticalScrollBarPolicy(QtCore.Qt.ScrollBarAsNeeded)
        self.current_filter.setObjectName(_fromUtf8("current_filter"))
        self.gridLayout.addWidget(self.current_filter, 1, 0, 1, 11)
        self.gridLayout.setRowStretch(0, 1)
        self.verticalLayout.addWidget(self.splitter)
        self.verticalLayout.setStretch(0, 1)

        self.retranslateUi(_Sqlite3TableExplorer)
        QtCore.QMetaObject.connectSlotsByName(_Sqlite3TableExplorer)

    def retranslateUi(self, _Sqlite3TableExplorer):
        _Sqlite3TableExplorer.setWindowTitle(_translate("_Sqlite3TableExplorer", "Dialog", None))
        self.second_sort_order.setItemText(0, _translate("_Sqlite3TableExplorer", "asc", None))
        self.second_sort_order.setItemText(1, _translate("_Sqlite3TableExplorer", "desc", None))
        self.first_sort_order.setItemText(0, _translate("_Sqlite3TableExplorer", "asc", None))
        self.first_sort_order.setItemText(1, _translate("_Sqlite3TableExplorer", "desc", None))
        self.reset_filter_button.setText(_translate("_Sqlite3TableExplorer", "reset filter", None))
        self.label.setText(_translate("_Sqlite3TableExplorer", "sort by", None))
        self.visible_columns_button.setText(_translate("_Sqlite3TableExplorer", "select columns", None))
        self.filter_button.setText(_translate("_Sqlite3TableExplorer", "set filter", None))
        self.label_2.setText(_translate("_Sqlite3TableExplorer", "sort by", None))

from eic_plotting_widget import EicPlottingWidget
from mz_plotting_widget import MzPlottingWidget
from widgets.choose_spectra_widget import ChooseSpectraWidget
from widgets.integration_widget import IntegrationWidget

class _Sqlite3TableExplorer(QtGui.QDialog, Ui__Sqlite3TableExplorer):
    def __init__(self, parent=None, f=QtCore.Qt.WindowFlags()):
        QtGui.QDialog.__init__(self, parent, f)

        self.setupUi(self)


if __name__ == "__main__":
    import sys
    app = QtGui.QApplication(sys.argv)
    _Sqlite3TableExplorer = QtGui.QDialog()
    ui = Ui__Sqlite3TableExplorer()
    ui.setupUi(_Sqlite3TableExplorer)
    _Sqlite3TableExplorer.show()
    sys.exit(app.exec_())

