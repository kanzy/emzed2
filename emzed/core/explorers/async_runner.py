# encoding: utf-8, division
from __future__ import print_function, division

from collections import defaultdict
import itertools

from .helpers import timethis


from PyQt4.QtCore import (QObject, QThread, Qt, pyqtSignal, pyqtSlot, QString, QTimer)


class Worker(QObject):

    finished = pyqtSignal()
    error = pyqtSignal(QString)
    result = pyqtSignal(object)

    def __init__(self, f, args):
        super(Worker, self).__init__()
        self.f = f
        self.args = args

    @pyqtSlot()
    def process(self):
        try:
            result = self.f(*self.args)
            self.result.emit(result)
        except Exception:
            import traceback
            e = traceback.format_exc()
            self.error.emit(e)
        finally:
            self.finished.emit()


class AsyncRunner(object):

    def __init__(self, parent):
        self.workers = defaultdict(list)
        self.parent = parent

    def print_error(self, e):
        print(e)

    def run_async_chained(self, functions, first_args):

        def start(i, args):
            if i >= len(functions):
                return None
            f = functions[i]
            def call_back(result):
                start(i + 1, (result,))
            self.run_async(f, args, call_back=call_back)

        start(0, first_args)

    def run_async(self, function, args, id_=None, only_one_worker=False, call_back=None,
                  blocked=False):

        if only_one_worker:
            for i, (thread, worker) in enumerate(self.workers[id_]):
                if thread.isRunning():
                    timethis(thread.quit)()
                    try:
                        del self.workers[id_][i]
                    except IndexError:
                        pass

        thread = QThread()
        worker = Worker(function, args)
        worker.moveToThread(thread)
        worker.error.connect(self.print_error)
        thread.started.connect(worker.process)
        worker.finished.connect(thread.quit)
        worker.finished.connect(thread.deleteLater)
        thread.finished.connect(worker.deleteLater)

        def unblock():
            self.parent.setCursor(Qt.ArrowCursor)

        thread.finished.connect(unblock)

        def set_waiting_cursor():
            self.parent.setCursor(Qt.WaitCursor)

        timer = QTimer()
        timer.setSingleShot(True)
        timer.timeout.connect(set_waiting_cursor)

        thread.finished.connect(timer.stop)
        timer.start(200)

        if call_back is not None:
            worker.result.connect(call_back)

        # we keep references, else the objects would get killed when the method
        # is finished, which crashs the application:
        self.workers[id_].append((thread, worker))

        def remove(workers=self.workers, id_=id_, thread=thread):
            while True:
                try:
                    # sometimes the c++ object already died, then we get
                    # RuntimeError: wrapped C/C++ object of type QThread has been deleted
                    running = thread.isRunning()
                except RuntimeError:
                    running = False
                if not running:
                    break
                time.sleep(.01)

            for i, (t, w) in itertools.izip(itertools.count(), self.workers[id_]):
                if t is thread:
                    try:
                        del workers[id_][i]
                    except IndexError:
                        pass  # might happen in race situations.

        thread.finished.connect(remove)
        thread.start()


if __name__ == "__main__":
    import  guidata
    app = guidata.qapplication()  # singleton !

    def f(x):
        return x + 1

    def g(x):
        print("got ", x)
        app.quit()

    runner = AsyncRunner()
    runner.run_async_chained((f, f,f,f,f,g), (0,))

    app.exec_()

