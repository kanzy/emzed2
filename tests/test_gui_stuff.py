# encoding: utf-8
from __future__ import print_function, division, absolute_import

import time

from emzed.core.explorers.async_runner import AsyncRunner

def test_async(regtest):

    import  guidata
    app = guidata.qapplication()  # singleton !

    def f(x):
        return x + 1

    def g(x):
        app.quit()

    def report(msg):
        print(msg, file=regtest)

    runner = AsyncRunner(reporter=report)
    runner.run_async_chained((f, f,f,f,f,g), (0,))
    app.exec_()
