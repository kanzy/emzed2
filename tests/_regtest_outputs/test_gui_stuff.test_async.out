run f (0,)
emitted 1
run f (1,)
emitted 2
run f (2,)
emitted 3
run f (3,)
emitted 4
run f (4,)
emitted 5
